src/main/java/services/RestoreFactory.java

```diff
@@ -110,8 +111,61 @@ private void handleAsyncRestoreResponse(String xml, HttpHeaders headers) {
         );
     }
 
-    public String getRestoreUrl(String domain){
-        log.info("Restoring from URL " + host + "/a/" + domain + "/phone/restore/?version=2.0");
+    private String getRestoreXmlHelper(String restoreUrl, HqAuth auth) {
+        RestTemplate restTemplate = new RestTemplate();
+        log.info("Restoring at domain: " + domain + " with auth: " + auth);
+        HttpHeaders headers = auth.getAuthHeaders();
+        headers.add("x-openrosa-version",  "2.0");
+        ResponseEntity<String> response = restTemplate.exchange(
+                restoreUrl,
+                HttpMethod.GET,
+                new HttpEntity<String>(headers),
+                String.class
+        );
+        if (response.getStatusCode().value() == 202) {
+            handleAsyncRestoreResponse(response.getBody(), response.getHeaders());
+        }
+        return response.getBody();
+    }
+
+    public static String getRestoreUrl(String host, String domain){
         return host + "/a/" + domain + "/phone/restore/?version=2.0";
     }
+
+    public String getRestoreUrl(String host, String domain, String username) {
+        return host + "/hq/admin/phone/restore/?as=" + username + "@" +
```

Comment:
we should use the `/a/<domain>/` route. i don't think this'll work for anyone who isn't a super usr